sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -554,8 +554,11 @@ class Analyzer(
           Cast(value, pivotColumn.dataType, Some(conf.sessionLocalTimeZone)).eval(EmptyRow)
         }
         // Group-by expressions coming from SQL are implicit and need to be deduced.
-        val groupByExprs = groupByExprsOpt.getOrElse(
-          (child.outputSet -- aggregates.flatMap(_.references) -- pivotColumn.references).toSeq)
+        val groupByExprs = groupByExprsOpt.getOrElse {
+          val pivotColAndAggRefs =
+            (pivotColumn.references ++ aggregates.flatMap(_.references)).toSet
```

Comment:
why do we need `toSet`?

Reply:
Otherwise the result type would be Iterable and we wouldn't be able to call "contains".

Reply:
well, then this is the problem, you are returning an Iterable instead of an `AttributeSet`. This can lead to wrong behaviors (eg. when the case of the attributes used is different) and cause a regression. I am submitting a followup for fixing this.